packages/cozy-harvest-lib/src/components/TriggerManager.jsx

```diff
@@ -235,18 +239,27 @@ export class TriggerManager extends Component {
   async launch(trigger) {
     const {
       launchTrigger,
+      onLaunch,
       onLoginSuccess,
       onSuccess,
       watchKonnectorJob
     } = this.props
     this.jobWatcher = watchKonnectorJob(await launchTrigger(trigger))
     this.jobWatcher.on('error', this.handleError)
-    this.jobWatcher.on('loginSuccess', () =>
-      this.handleSuccess(onLoginSuccess, [trigger])
-    )
-    this.jobWatcher.on('success', () =>
-      this.handleSuccess(onSuccess, [trigger])
-    )
+    this.jobWatcher.on('loginSuccess', () => this.handleSuccess(onLoginSuccess))
+    this.jobWatcher.on('success', () => this.handleSuccess(onSuccess))
```

Comment:
A watcher seems to be created on every `launch` call, and the attached events don't seem to be detached. Shouldn't we detach it after error or success?

Reply:
We could. Actually each event is assumed to occur once : there will be no `error` after `success` and no `success` after `error`.

By the way, we still want to merge this part with the KonnectorJob model. Are you ok to do this in a next PR ?

Reply:
I was also thinking about potential memory leak. Maybe add a TODO comment if you can't fix this now.

Reply:
https://github.com/cozy/cozy-libs/issues/570 🙏